lib/methods/veres-one/veres-did-doc.js

```diff
@@ -240,6 +240,106 @@ class VeresOneDidDoc {
     };
   }
 
+  /**
+   * Composes and returns a service id for a service name.
+   *
+   * @param serviceName {string}
+   *
+   * @returns {string} Service id
+   */
+  serviceIdFor(serviceName) {
+    if(!serviceName) {
+      throw new Error('Invalid service name');
+    }
+
+    return this.id + ';' + encodeURIComponent(serviceName);
+  }
+
+  /**
+   * Finds a service endpoint in this did doc, given an id or a name.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   *
+   * @returns {object}
+   */
+  findService({id, name}) {
+    const jsonld = this.injector.use('jsonld');
+
+    const serviceId = id || this.serviceIdFor(name);
+
+    return jsonld
+      .getValues(this.doc, 'service')
+      .find(service => service.id === serviceId);
+  }
+
+  /**
+   * Tests whether this did doc has a service endpoint (by name or id).
+   * One of `id` or `name` is required.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   *
+   * @returns {boolean}
+   */
+  hasService({id, name}) {
+    return !!this.findService({id, name});
+  }
+
+  /**
+   * Adds a service endpoint to this did doc.
+   * One of `id` or `name` is required.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   * @param type {string} Endpoint type (e.g. 'AgentService')
+   * @param endpoint {string} Endpoint uri (e.g. 'https://agent.example.com')
+   * @param [options] {object} Any additional properties of endpoint
+   */
+  addService({id, name, type, endpoint, ...options}) {
```

Comment:
This API is a bit odd.  The docs above imply you pass in an optional options object, but the code is just slurping in all extra params into options.  That design does make it impossible to include `name` or `endpoint` in the endpoint object.  And you could cause madness by passing in different `endpoint` and `serviceEndpoint` values.  I'm not sure what to suggest here, but at least clear up the docs.

Reply:
Apologies, I think the problem is just in the jsdoc (I wasn't sure how to denote `...options` in its syntax). 
The idea I was going for is - you should be able to pass in arbitrary properties for the node, yeah?

Reply:
Yeah. I'm not sure how to do that in jsdoc.  The edge case of using extra properties named `endpoint` or `name` could be solved with an actual options object.  Anyone else have thoughts on what this API should look like and allow?

Reply:
Open to suggestions on this (or we can open an issue to revisit it..?)